Firestore/core/src/util/executor_std.h

```diff
@@ -62,24 +64,29 @@ class ExecutorStd : public Executor {
   Task* PopFromSchedule() override;
 
  private:
-  Id PushOnSchedule(TimePoint when, Tag tag, Operation&& operation);
+  class SharedState;
+
+  Id PushOnScheduleLocked(TimePoint when, Tag tag, Operation&& operation);
 
   void OnCompletion(Task* task) override;
   void Cancel(Id operation_id) override;
 
-  void PollingThread();
-  Id NextId();
+  static void PollingThread(std::shared_ptr<SharedState> state);
+  Id NextIdLocked();
```

Comment:
I liked the pattern used in `Task` where `*Locked` functions would take a `unique_lock` as the first argument. Can the same be done here?

Reply:
I only did that for the one method in `Task::AwaitLocked` and did so specifically because that method needed to wait on a condition variable. The `Locked` suffix alone is enough of an indication that the lock needs to be held so I'm wary of also requiring that the lock be passed even when the method isn't going to use it.